lib/elixir/lib/enum.ex

```diff
@@ -2030,6 +2030,8 @@ defmodule Enum do
   Reverses the elements in `enumerable`, appends the `tail`, and returns
   it as a list.
 
+  If `tail` is an improper-list, it returns an improper-list.
```

Comment:
This is the same as #8469. We expect an enumerable here and an improper list is not an enumerable. It works only for optimization purposes but it wouldn't work like if the language was typed, for example.